### PR TITLE
Foreign Recipient Location Fix

### PR DESCRIPTION
**Pull request by justinle**
**Description:**
Removed instances of United States appearing in results while "Foreign Countries" is checked.

**Requirements for PR merge:**

1. [x] Necessary PR reviewers:
	- [x] Backend
2. [x] Unit tests updated
8. [x] Jira Ticket [DEV-1642](https://federal-spending-transparency.atlassian.net/browse/DEV-1642):
	- [x] Link to this Pull-Request

### CODE REVIEW

**Comment by IsaacRay** [comment]
Can you post some performance metrics (i.e., before/after api request timings)? Thanks.

File: usaspending_api/search/tests/test_spending_by_award_type_count.py [hunk 1]
```diff
@@ -1,128 +0,0 @@
-import json
```
[the remark this answers is not in the document]
Comment (justinle) [reply usaspending_api/search/tests/test_spending_by_award_type_count.py]: My mistake, I meant to delete an obsolete test file I made. File is restored on most recent commit.

[the remark this answers is not in the document]
Comment (tony-sappe) [reply usaspending_api/search/tests/test_spending_by_award_type_count.py]: Looks like commit 1820a16 was unsuccessful. 
Try `git checkout origin/dev -- usaspending_api/search/tests/test_spending_by_award_type_count.py`

**Comment by justinle** [comment]
Metrics performed through the `spending_by_award` endpoint between dev branch and this branch showed a negligible difference in changes. Performing the test using both domestic, foreign averaged ~1500ms. Two separate sessions of each test were performed, and it may be noteworthy that the averages seemed to be more dependent on the session than the branch changes.

File: usaspending_api/search/tests/test_spending_by_award_type.py [hunk 2]
```diff
@@ -237,3 +237,54 @@ def test_spending_by_award_both_zip_filter(client, mock_matviews_qs):
...
+
+
+@pytest.mark.django_db
+def test_spending_by_award_foreign_filter(client, mock_matviews_qs):
+    """ Verify that foreign country filter is returning the correct results """
+    mock_model_1 = MockModel(award_id=1, piid=None, fain='abc', uri=None,
+                             type='B', pulled_from="AWARD", recipient_location_country_name="United States",
+                             recipient_location_country_code="USA")
+    mock_model_2 = MockModel(award_id=2, piid=None, fain='abd', uri=None,
+                             type='B', pulled_from="AWARD", recipient_location_country_name="United States",
```
Comment (IsaacRay) [inline usaspending_api/search/tests/test_spending_by_award_type.py]: Is this ("United States") definitely lowercase? In the place of performance filter, its all uppercased...

File: usaspending_api/search/tests/test_spending_by_award_type.py [hunk 3]
```diff
@@ -237,3 +237,54 @@ def test_spending_by_award_both_zip_filter(client, mock_matviews_qs):
...
+                    "C",
+                    "D"
+                ],
+                "recipient_scope": "foreign"
+            },
+            "fields": [
+                "Award ID"
+            ],
+        }))
+    assert len(resp.data['results']) == 1
```
Comment (IsaacRay) [inline usaspending_api/search/tests/test_spending_by_award_type.py]: You should also include an assertion verifying that "recipient_scope: domestic" returns 2 results.

File: usaspending_api/search/tests/test_spending_by_award_type.py [hunk 4]
```diff
@@ -237,3 +237,54 @@ def test_spending_by_award_both_zip_filter(client, mock_matviews_qs):
         }))
     assert len(resp.data['results']) == 1
     assert resp.data['results'][0] == {'internal_id': 1, 'Place of Performance Zip5': '00001'}
+
+
+@pytest.mark.django_db
+def test_spending_by_award_foreign_filter(client, mock_matviews_qs):
+    """ Verify that foreign country filter is returning the correct results """
+    mock_model_1 = MockModel(award_id=1, piid=None, fain='abc', uri=None,
```
Comment (IsaacRay) [inline usaspending_api/search/tests/test_spending_by_award_type.py]: Also, one of these mock models should have JUST a recipient_location_country_name, and the other should have JUST a recipient_loaction_country_code... you want to make sure your query returns both.

File: usaspending_api/search/tests/test_spending_by_award_type.py (hunk 3 above)
Comment (justinle) [reply usaspending_api/search/tests/test_spending_by_award_type.py]: Added additional tests that account for country code only, country name only, and country name and country code.

**Pull request merged**